File: ios/Superwall.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(Superwall, RCTEventEmitter)

RCT_EXTERN_METHOD(initPaywall:(NSString)superwallApiKey revenueCatApiKey:(NSString)revenueCatApiKey)

RCT_EXTERN_METHOD(setRegister:(NSString)isRegister)

RCT_EXTERN_METHOD(identify:(NSString)userId)

RCT_EXTERN_METHOD(reset)

RCT_EXTERN_METHOD(trigger:(NSString)campaignName resolver: (RCTPromiseResolveBlock)resolve rejecter: (RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/Superwall.swift
import Foundation
import Paywall

@objc(Superwall)
class Superwall: RCTEventEmitter {
    
    public static var emitter: RCTEventEmitter!

      override init() {
        super.init()
        Superwall.emitter = self
      }

      open override func supportedEvents() -> [String] {
        ["superwallAnalyticsEvent"]
      }
    
    
    @objc(initPaywall:revenueCatApiKey:)
    func initPaywall(superwallApiKey:String, revenueCatApiKey:String) -> Void {
        PaywallService.initPaywall(superwallApiKey: superwallApiKey, revenueCatApiKey: revenueCatApiKey)
    }
    
    @objc(setRegister:)
    func setRegister(isRegister:String) -> Void {
    let attributes: [String: Any] = [
          "register": isRegister,
          "testing": "false"
        ]
        Paywall.setUserAttributes(attributes)
    }

    @objc(identify:)
    func identify(userId:String) -> Void {
        Paywall.identify(userId: userId)
    }

    @objc(reset:)
    func reset(test:String) -> Void {
        Paywall.reset()
    }

    @objc(trigger:resolver:rejecter:)
    func trigger(campaignName:String, resolver: @escaping RCTPromiseResolveBlock,  rejecter: @escaping RCTPromiseRejectBlock) -> Void {
        
        Paywall.trigger(
          event: campaignName,
          params: nil,
          onSkip: { error in
            rejecter("error", error?.description, nil)
          },
          onPresent: nil,
          onDismiss: { didPurchase, productId, paywallInfo in
              if didPurchase {
                  resolver(["didPurchase": true, "productId": productId])
              } else {
                resolver(["didPurchase": false, "productId": nil])
              }
            }
        )
    }

    
}
